builtin/logical/pki/path_fetch_keys.go

```diff
@@ -0,0 +1,253 @@
+package pki
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/vault/sdk/framework"
+	"github.com/hashicorp/vault/sdk/logical"
+)
+
+func pathListKeys(b *backend) *framework.Path {
+	return &framework.Path{
+		Pattern: "keys/?$",
+
+		Operations: map[logical.Operation]framework.OperationHandler{
+			logical.ListOperation: &framework.PathOperation{
+				Callback:                    b.pathListKeysHandler,
+				ForwardPerformanceStandby:   false,
+				ForwardPerformanceSecondary: false,
+			},
+		},
+
+		HelpSynopsis:    pathListKeysHelpSyn,
+		HelpDescription: pathListKeysHelpDesc,
+	}
+}
+
+const (
+	pathListKeysHelpSyn  = `Fetch a list of all issuer keys`
+	pathListKeysHelpDesc = `This endpoint allows listing of known backing keys, returning
+their identifier and their name (if set).`
+)
+
+func (b *backend) pathListKeysHandler(ctx context.Context, req *logical.Request, _ *framework.FieldData) (*logical.Response, error) {
+	if b.useLegacyBundleCaStorage() {
+		return logical.ErrorResponse("Can not list keys until migration has completed"), nil
+	}
+
+	var responseKeys []string
+	responseInfo := make(map[string]interface{})
+
+	entries, err := listKeys(ctx, req.Storage)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, identifier := range entries {
+		key, err := fetchKeyById(ctx, req.Storage, identifier)
+		if err != nil {
+			return nil, err
+		}
+
+		responseKeys = append(responseKeys, string(identifier))
+		responseInfo[string(identifier)] = map[string]interface{}{
```

Comment:
I don't know if it is too late for this, but it seems like we could simplify the output of this endpoint a little with something like (paraphrased):

```json
{"keys": [
  {"id": "uuid...", other info ... },
  {"id": "uuid2...", other info ... },
]}
```  
 
Mostly, I'm always hesitant to put semantic meaning in the keys of a map (i.e., the keyIds being present as keys), when really what you have is a list. Plus, this means it is easier in the future to have multiple entries with the same `id` (like, if you had previous versions, or multiple valid versions, etc.)

Reply:
That's sadly the format of `ListResponseWithInfo` that we've gotta obey. Its a standard Vault Response type, :/

Reply:
:+1: